msvc: Update vcpkg manifest

hebasto opened the pull request:
This PR updates the vcpkg manifest baseline from the [2023.08.09 Release ](https://github.com/microsoft/vcpkg/releases/tag/2023.08.09) to the [2024.09.30 Release](https://github.com/microsoft/vcpkg/releases/tag/2024.09.30), with the following package changes:
- `boost`: 1.82.0#2 --> 1.85.0#1,2
- `qt5`: 5.15.10#5 -> 5.15.15
- `sqlite3`: 3.42.0#1 --> 3.46.1
- `zeromq`: 2023-06-20#1 --> 4.3.5#2

The previous update was made in https://github.com/bitcoin/bitcoin/pull/28938.

For additional minor improvements, please refer to the commit messages.

hebasto commented:
Friendly ping @sipsorcery @hodlinator ;)

hodlinator reviewed (commented):
Tested on Windows 11 Home:
```
> rmdir /S build
> cmake -B build --preset vs2022-static
> cmake --build build --config Release -j<X>
> build\src\qt\Release\bitcoin-qt.exe (started successfully)
```

hebasto commented on vcpkg.json:
```diff
@@ -1,46 +1,64 @@
-    {"name": "libevent", "version": "2.1.12#7"},
-    {"name": "liblzma", "version": "5.4.1#1"}
-  ],
+  "$comment": "The builtin-baseline corresponds to 2024.09.30 Release",
```
This `"$comment"` field actually comments the following line:https://github.com/bitcoin/bitcoin/blob/fad685ce54d39a333a6c75641b5d519cb0345959/vcpkg.json#L3-L4

hodlinator replied:
Saw that, but it mentions it by name so I thought maybe keeping the lexicographical order the first commit tries to introduce would have precedent.

hebasto committed:
build, msvc: Reorder keys in `vcpkg.json`

This change ensures that any further manipulation of `vcpkg.json` via
the `vcpkg` command will produce minimal diffs.

hebasto committed:
build, msvc: Drop no longer needed `liblzma` version pinning

For historical context, see https://github.com/hebasto/bitcoin/pull/140.

hebasto committed:
build, msvc: Document `libevent` version pinning

hebasto committed:
build, msvc: Update vcpkg manifest baseline

The `"$comment"` field is positioned at the top, as it will be
automatically placed there during any processing of this file by
`vcpkg.exe`.

hebasto commented on vcpkg.json:
I've dropped mentioning of lexicographical order from the commit message. The goal of the reordering is to minimize diffs for invocations like `vcpkg add port libqrencode`.

fanquake reviewed (approved):
ACK f6577b717416191e65ba8221be57f1a48d74e5d7
